visualization/app/codeCharta/ui/codeMap/codeMap.arrow.service.ts

```diff
@@ -23,25 +23,41 @@ export class CodeMapArrowService implements BuildingHoveredSubscriber, BuildingU
 		this.arrows = new Array<Object3D>()
 		CodeMapMouseEventService.subscribeToBuildingHovered(this.$rootScope, this)
 		CodeMapMouseEventService.subscribeToBuildingUnhovered(this.$rootScope, this)
+		ThreeSceneService.subscribeToBuildingSelectedEvents(this.$rootScope, this)
+		ThreeSceneService.subscribeToBuildingDeselectedEvents(this.$rootScope, this)
+	}
+
+	public onBuildingSelected(selectedBuilding: CodeMapBuilding) {
+		const settings = this.settingsService.getSettings()
+		if (settings.dynamicSettings.edgeMetric !== "None" && !selectedBuilding.node.flat) {
+			this.clearArrows()
+			this.showEdgesOfBuildings(settings.fileSettings.edges)
+		}
+		this.scale(this.settingsService.getSettings().appSettings.scaling)
+	}
+
+	public onBuildingDeselected() {
+		const settings = this.settingsService.getSettings()
```

Comment:
declaring a new variable with the same name that is only used once after is not really necessary. just use
```
this.addEdgePreview(null, this.settingsService.getSettings().fileSettings.edges.filter(x => x.visible != EdgeVisibility.none))
```

below

Reply:
Sometimes it's fine. Especially when the line gets really long, but I don't think that's the case here